From b8a3defdabc0c36a0b7090ffae713d45cd590436 Mon Sep 17 00:00:00 2001
From: bgill55 <bricam55@gmail.com>
Date: Sat, 24 Jan 2015 18:47:24 -0700
Subject: [PATCH] Exposing hard coded resources for type 2 overlay access [5/6]

Have to expose hard coded hex coding to changeable resources for theming purposes. For the Layers type 2 overlays.

Change-Id: I5039c2ad6fb765a94285123d3c0566132b3bf9ac

Conflicts:
	res/values/styles.xml
---
 res/drawable/floating_action_button.xml         |  2 +-
 res/drawable/recent_lists_footer_background.xml |  2 +-
 res/layout/call_detail.xml                      |  2 +-
 res/layout/dialpad_fragment.xml                 |  2 +-
 res/layout/keyguard_preview.xml                 |  2 +-
 res/values/colors.xml                           | 23 +++++++++++++
 res/values/styles.xml                           | 44 +++++++++++++++----------
 7 files changed, 54 insertions(+), 23 deletions(-)

diff --git a/res/drawable/floating_action_button.xml b/res/drawable/floating_action_button.xml
index d550190..a1088b2 100644
--- a/res/drawable/floating_action_button.xml
+++ b/res/drawable/floating_action_button.xml
@@ -19,7 +19,7 @@
     android:color="@color/floating_action_button_touch_tint">
     <item android:id="@android:id/mask">
         <shape android:shape="oval">
-            <solid android:color="@android:color/white" />
+            <solid android:color="@color/exposed_floating_action_button_solid" />
         </shape>
     </item>
 </ripple>
\ No newline at end of file
diff --git a/res/drawable/recent_lists_footer_background.xml b/res/drawable/recent_lists_footer_background.xml
index b5029af..b813bd1 100644
--- a/res/drawable/recent_lists_footer_background.xml
+++ b/res/drawable/recent_lists_footer_background.xml
@@ -19,6 +19,6 @@
     android:color="?android:attr/colorControlHighlight">
     <!-- Mask to constrain the ripple to the bounds of the view. -->
     <item android:id="@android:id/mask">
-        <color android:color="@android:color/white" />
+        <color android:color="@color/exposed_recents_list_footer_bg" />
     </item>
 </ripple>
diff --git a/res/layout/call_detail.xml b/res/layout/call_detail.xml
index 35c41c0..b4d6065 100644
--- a/res/layout/call_detail.xml
+++ b/res/layout/call_detail.xml
@@ -108,7 +108,7 @@
         android:id="@+id/blank"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:background="@android:color/black"
+        android:background="@color/exposed_bg_light"
         android:visibility="gone"
         android:clickable="true"
         android:layout_alignParentStart="true"
diff --git a/res/layout/dialpad_fragment.xml b/res/layout/dialpad_fragment.xml
index 21cb586..5905a4e 100644
--- a/res/layout/dialpad_fragment.xml
+++ b/res/layout/dialpad_fragment.xml
@@ -30,7 +30,7 @@
             android:layout_width="match_parent"
             android:layout_height="0dp"
             android:layout_weight="1"
-            android:background="#00000000" />
+            android:background="@color/exposed_dialpad_fragment" />
         <!-- Dialpad shadow -->
         <View
             android:layout_width="match_parent"
diff --git a/res/layout/keyguard_preview.xml b/res/layout/keyguard_preview.xml
index 16243eb..e8dc6fd 100644
--- a/res/layout/keyguard_preview.xml
+++ b/res/layout/keyguard_preview.xml
@@ -26,5 +26,5 @@
         android:layout_width="match_parent"
         android:layout_weight="1"
         android:layout_height="0dp"
-        android:background="#ffffff" />
+        android:background="@color/exposed_bg_dark" />
 </LinearLayout>
diff --git a/res/values/colors.xml b/res/values/colors.xml
index 0a1552f..cea69fa 100644
--- a/res/values/colors.xml
+++ b/res/values/colors.xml
@@ -97,5 +97,28 @@
     <color name="dialer_dialpad_touch_tint">#330288d1</color>
 
     <color name="floating_action_button_touch_tint">#80ffffff</color>
+    
+    
+    <!-- Exposed resources for theming -->
+    <color name="exposed_bg_dark">#ffffffff</color>
+    <color name="exposed_bg_light">#ff000000</color>
+    <color name="exposed_primary_text_light">#de000000</color>
+    <color name="exposed_secondary_text_light">#8a000000</color>
+    <color name="exposed_primary_text_dark">#ffffffff</color>
+    <color name="exposed_secondary_text_dark">#b3ffffff</color>
+    <color name="exposed_secondary_bg_color">#333333</color>
+    <color name="exposed_call_log_header">#33b5e5</color>
+    <color name="exposed_call_log_voicemail_status_background">#262626</color>
+    <color name="exposed_call_log_voicemail_status_text">#888888</color>
+    <color name="exposed_call_log_voicemail_status_action_text">#33b5e5</color>
+    <color name="exposed_floating_action_button_solid">#ffffffff</color>
+    <color name="exposed_recents_list_footer_bg">#ffffffff</color>
+    <color name="exposed_dialpad_fragment">#00000000</color>
+    
+    
+    
+    
+    
+    
 
 </resources>
diff --git a/res/values/styles.xml b/res/values/styles.xml
index fbb400f..1f8be68 100644
--- a/res/values/styles.xml
+++ b/res/values/styles.xml
@@ -62,22 +62,25 @@
         <item name="list_item_text_offset_top">-2dp</item>
         <!-- CallLog -->
         <item name="call_log_primary_text_color">@color/dialtacts_primary_text_color</item>
-        <item name="call_log_primary_background_color">#000000</item>
+	<!-- exposed bg for theming  -->
+        <item name="call_log_primary_background_color">@color/exposed_bg_light</item>
         <item name="call_log_secondary_text_color">@color/dialtacts_secondary_text_color</item>
-        <item name="call_log_secondary_background_color">#333333</item>
-        <item name="call_log_header_color">#33b5e5</item>
+        <item name="call_log_secondary_background_color">@color/exposed_secondary_bg_color</item>
+        <item name="call_log_header_color">@color/exposed_call_log_header</item>
         <!-- VoicemailStatus -->
         <item name="call_log_voicemail_status_height">48dip</item>
-        <item name="call_log_voicemail_status_background_color">#262626</item>
-        <item name="call_log_voicemail_status_text_color">#888888</item>
-        <item name="call_log_voicemail_status_action_text_color">#33b5e5</item>
+	<!-- exposed textcolor and bg for theming  -->
+        <item name="call_log_voicemail_status_background_color">@color/exposed_call_log_voicemail_status_background</item>
+        <item name="call_log_voicemail_status_text_color">@color/exposed_call_log_voicemail_status_text</item>
+        <item name="call_log_voicemail_status_action_text_color">@color/exposed_call_log_voicemail_status_action_text</item>
             <!-- Favorites -->
         <item name="favorites_padding_bottom">?android:attr/actionBarSize</item>
-        <item name="android:colorPrimary">@color/dialer_theme_color</item>
-        <item name="android:colorPrimaryDark">@color/dialer_theme_color_dark</item>
+	<!-- exposed textcolor for theming  -->
+        <item name="android:colorPrimary">@color/exposed_primary_text_dark</item>
+        <item name="android:colorPrimaryDark">@color/exposed_secondary_text_light</item>
         <item name="android:navigationBarColor">@color/dialer_theme_color_dark</item>
         <item name="dialpad_key_button_touch_tint">@color/dialer_dialpad_touch_tint</item>
-        <item name="android:colorControlActivated">@color/dialer_theme_color</item>
+        <item name="android:colorControlActivated">@color/exposed_primary_text_dark</item>
     </style>
 
     <!-- Action bar overflow menu icon. -->
@@ -93,7 +96,8 @@
     </style>
 
     <style name="DialpadTheme" parent="DialtactsTheme">
-        <item name="android:textColorPrimary">#FFFFFF</item>
+      <!-- exposed textcolor for theming  -->
+        <item name="android:textColorPrimary">@color/exposed_primary_text_dark</item>
     </style>
 
     <style name="DialtactsThemeWithoutActionBarOverlay" parent="DialtactsTheme">
@@ -111,14 +115,16 @@
     <style name="CallDetailActivityTheme" parent="DialtactsThemeWithoutActionBarOverlay">
         <item name="android:windowBackground">@color/background_dialer_results</item>
         <!-- CallLog -->
-        <item name="call_log_primary_background_color">#FFFFFF</item>
-        <item name="call_log_secondary_background_color">#FFFFFF</item>
-        <item name="call_log_header_color">#FFFFFF</item>
+	<!-- exposed  bg for theming  -->
+        <item name="call_log_primary_background_color">@color/exposed_bg_dark</item>
+        <item name="call_log_secondary_background_color">@color/exposed_bg_dark</item>
+        <item name="call_log_header_color">@color/exposed_bg_dark</item>
         <!-- VoicemailStatus -->
+	<!-- exposed bg for theming  -->
         <item name="call_log_voicemail_status_height">48dip</item>
-        <item name="call_log_voicemail_status_background_color">#262626</item>
-        <item name="call_log_voicemail_status_text_color">#888888</item>
-        <item name="call_log_voicemail_status_action_text_color">#33b5e5</item>
+        <item name="call_log_voicemail_status_background_color">@color/exposed_call_log_voicemail_status_background</item>
+        <item name="call_log_voicemail_status_text_color">@color/exposed_call_log_voicemail_status_text</item>
+        <item name="call_log_voicemail_status_action_text_color">@color/exposed_call_log_voicemail_status_action_text</item>
         <item name="android:actionOverflowButtonStyle">@style/DialtactsActionBarOverflowWhite</item>
     </style>
 
@@ -144,13 +150,15 @@
     <!-- Text in the action bar at the top of the screen -->
     <style name="DialtactsActionBarTitleText"
            parent="@android:style/TextAppearance.Material.Widget.ActionBar.Title">
-        <item name="android:textColor">@color/actionbar_text_color</item>
+      <!-- exposed textcolor for theming  -->
+        <item name="android:textColor">@color/exposed_primary_text_dark</item>
     </style>
 
     <!-- Text style for tabs. -->
     <style name="DialtactsActionBarTabTextStyle"
            parent="android:style/Widget.Material.Light.ActionBar.TabText">
-        <item name="android:textColor">@color/tab_text_color</item>
+      <!-- exposed textcolor for theming  -->
+        <item name="android:textColor">@color/exposed_primary_text_dark</item>
         <item name="android:textSize">@dimen/tab_text_size</item>
         <item name="android:fontFamily">"sans-serif-medium"</item>
     </style>
-- 
1.9.1

